testgrid/conformance/upload_e2e.py

```diff
@@ -0,0 +1,206 @@
+    # contain one and we want a datetime object...
+    timestamp = year+' '+match.group(1)
+    return datetime.strptime(timestamp, '%Y %b %d %H:%M:%S.%f')
+
+def parse_e2e_logfile(f, year):
+    """parse e2e logfile at path, assuming the log is from year
+    
+    Args:
+        f (file): the log file, iterated for lines
+        year (str): YYYY year logfile is from
+
+    Returns:
+        started (datetime.datetime), finished (datetime.datetime), passed (boolean)
+    """
+    started = finished = None
+    passed = False
+    for line in f:
+        # try to get a timestamp from each line, keep the first one as
+        # start time, and the last one as finish time
+        ts = parse_e2e_log_line_timestamp(line, year)
+        if started is None:
+            started = ts
+        elif ts:
+            finished = ts
+        # if we found the ginkgo success line then the run passed
+        is_success = E2E_LOG_SUCCESS_RE.match(line)
+        if is_success:
+            passed = True
+    return started, finished, passed
+
+def datetime_to_unix(dt):
+    """convert datetime.datetime to unix timestamp"""
+    return int(time.mktime(dt.timetuple()))
+
+def testgrid_started_json_contents(start_time):
+    """returns the string contents of a testgrid started.json file
+
+    Args:
+        start_time (datetime.datetime)
+
+    Returns:
+        contents (str)
+    """
+    started = datetime_to_unix(start_time)
+    return json.dumps({
+        "timestamp": started
+    })
+
+def testgrid_finished_json_contents(finish_time, passed):
+    """returns the string contents of a testgrid finished.json file
+
+    Args:
+        finish_time (datetime.datetime)
+        passed (bool)
+
+    Returns:
+        contents (str)
+    """
+    finished = datetime_to_unix(finish_time)
+    result = "SUCCESS" if passed else "FAILURE"
+    return json.dumps({
+        "timestamp": finished,
+        "result": result
+    })
+
+def upload_string(gcs_path, text):
+    """Uploads text to gcs_path"""
+    cmd = ['gsutil', '-q', '-h', 'Content-Type:text/plain', 'cp', '-', gcs_path]
+    print >>sys.stderr, 'Run:', cmd, 'stdin=%s'%text
+    proc = subprocess.Popen(cmd, stdin=subprocess.PIPE)
+    proc.communicate(input=text)
+    if proc.returncode != 0:
+        raise RuntimeError("Failed to upload with exit code: %d" % proc.returncode)
+
+def upload_file(gcs_path, file_path):
+    """Uploads file at file_path to gcs_path"""
+    cmd = ['gsutil', '-q', '-h', 'Content-Type:text/plain', 'cp', file_path, gcs_path]
+    print >>sys.stderr, 'Run:', cmd
+    proc = subprocess.Popen(cmd)
+    proc.communicate()
+    if proc.returncode != 0:
+        raise RuntimeError("Failed to upload with exit code: %d" % proc.returncode)
+
+def parse_args(cli_args=None):
+    if cli_args is None:
+        cli_args = sys.argv[1:]
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--bucket', 
+        help='GCS bucket to upload the results to,' +\
```

Comment:
I would instead do:
```
  help=('GCS bucket to '
            'upload to')
```

Rather than concatenating

https://google.github.io/styleguide/pyguide.html?showone=Line_length#Line_length

Reply:
done